src/main/java/org/broadinstitute/consent/http/authentication/OAuthAuthenticator.java

```diff
@@ -1,112 +1,41 @@
 package org.broadinstitute.consent.http.authentication;
 
-import com.google.gson.Gson;
 import com.google.inject.Inject;
 import io.dropwizard.auth.Authenticator;
 import jakarta.ws.rs.NotFoundException;
 import jakarta.ws.rs.ServerErrorException;
-import jakarta.ws.rs.WebApplicationException;
-import java.util.Map;
-import java.util.Objects;
 import java.util.Optional;
-import org.broadinstitute.consent.http.filters.ClaimsCache;
 import org.broadinstitute.consent.http.models.AuthUser;
-import org.broadinstitute.consent.http.models.sam.UserStatus;
-import org.broadinstitute.consent.http.models.sam.UserStatusInfo;
+import org.broadinstitute.consent.http.models.DuosUser;
+import org.broadinstitute.consent.http.models.User;
+import org.broadinstitute.consent.http.service.UserService;
 import org.broadinstitute.consent.http.service.sam.SamService;
-import org.broadinstitute.consent.http.util.ConsentLogger;
 
-
-public class OAuthAuthenticator implements Authenticator<String, AuthUser>, ConsentLogger {
-
-  private final SamService samService;
-  private final ClaimsCache claimsCache;
+public class OAuthAuthenticator extends AuthenticatorHelper implements
+    Authenticator<String, AuthUser> {
 
   @Inject
-  public OAuthAuthenticator(SamService samService) {
-    this.samService = samService;
-    this.claimsCache = ClaimsCache.getInstance();
+  public OAuthAuthenticator(SamService samService, UserService userService) {
+    super(samService, userService);
   }
 
   @Override
   public Optional<AuthUser> authenticate(String bearer) {
     var headers = claimsCache.cache.getIfPresent(bearer);
     if (headers != null) {
-      AuthUser user = buildAuthUserFromHeaders(headers);
-      AuthUser userWithStatus = getUserWithStatusInfo(user);
-      if (userWithStatus == null) {
-        logWarn("User with status is null, authentication incomplete");
-        return Optional.of(user);
+      AuthUser authUser = buildAuthUserFromHeaders(headers);
+      if (authUser.getEmail() != null) {
+        authUser.setUserStatusInfo(getUserStatusInfo(authUser));
+        try {
+          User duosUser = userService.findUserByEmail(authUser.getEmail());
+          return Optional.of(new DuosUser(authUser, duosUser));
+        } catch (NotFoundException e) {
+          logWarn("User not found, authentication incomplete: %s".formatted(authUser.getEmail()));
+        }
```

Comment:
Is this needed now? If a `User` is required, you can use `DuosUser`. If it's not, you can use `AuthUser`. Until the code migrates to using `DuosUser`, we can continue to look up `User` manually on API calls.

Reply:
Good point - we don't need this here any longer